village/src/components/SmurfForm.js

```diff
@@ -1,55 +1,99 @@
-import React, { Component } from 'react';
-
+import React, { Fragment, Component } from "react";
+import axios from "axios";
+import { Link } from 'react-router-dom';
 class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      name: '',
-      age: '',
-      height: ''
+      name: "",
+      age: "",
+      height: ""
     };
   }
 
   addSmurf = event => {
     event.preventDefault();
     // add code to create the smurf using the api
-
+    axios
+      .post(`http://localhost:3333/smurfs`, { ...this.state })
+      .then(res => {
+        this.setState({
+          smurfs: res.data
+        });
```

Comment:
Here you are setting the state on SmurfForm.js, but not in App.js and is the reason your added smurfs aren't rendering until refresh